src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -0,0 +1,267 @@
+                    score = get_complexity_score(child.cumulative_node_complexity)
+                    if self._is_node_valid_to_breakdown(score, child):
+                        # Append score and child to the pipeline breaker sorted list
+                        # so that the valid child with the highest complexity score
+                        # is at the end of the list.
+                        pipeline_breaker_list.add((score, child))
+                    else:
+                        # don't traverse subtrees if parent is a valid candidate
+                        next_level.append(child)
+
+            current_level = next_level
+
+        if not pipeline_breaker_list:
+            # Return None if no valid node is found for partitioning.
+            return None
+
+        # Get the node with the highest complexity score
+        _, child = pipeline_breaker_list.pop()
+
+        # Create a temp table for the partitioned node
+        temp_table_name = self._get_temp_table_name()
+        temp_table_plan = self._query_generator.resolve(
+            SnowflakeCreateTable(
+                [temp_table_name],
+                None,
+                SaveMode.ERROR_IF_EXISTS,
+                child,
+                table_type="temp",
+                creation_source=TableCreationSource.LARGE_QUERY_BREAKDOWN,
+            )
+        )
+
+        # Update the ancestors with the temp table selectable
+        self._replace_child_and_update_ancestors(parent_map, child, temp_table_name)
+
+        return temp_table_plan
+
+    def _is_node_valid_to_breakdown(self, score: int, node: LogicalPlan) -> bool:
+        """Method to check if a node is valid to breakdown based on complexity score and node type."""
+        return (
+            COMPLEXITY_SCORE_LOWER_BOUND < score < COMPLEXITY_SCORE_UPPER_BOUND
+        ) and self._is_node_pipeline_breaker(node)
+
+    def _is_node_pipeline_breaker(self, node: LogicalPlan) -> bool:
+        """Method to check if a node is a pipeline breaker based on the node type.
+
+        If the node contains a SnowflakePlan, we check its source plan recursively.
+        """
+        if isinstance(node, (Pivot, Unpivot, Sort, Aggregate)):
+            return True
+
+        if isinstance(node, Sample):
+            return node.row_count is not None
+
+        if isinstance(node, Union):
+            return not node.is_all
+
+        if isinstance(node, SelectStatement):
+            return node.order_by is not None
+
+        if isinstance(node, SetStatement):
+            return any(operand.operator == SET_UNION for operand in node.set_operands)
+
+        if isinstance(node, SnowflakePlan):
+            return node.source_plan is not None and self._is_node_pipeline_breaker(
+                node.source_plan
+            )
+
+        if isinstance(node, (SelectSnowflakePlan, SelectTableFunction)):
+            return self._is_node_pipeline_breaker(node.snowflake_plan)
```

Comment:
if it is a SelectTableFunction, we know the source node will be a TableFunctionJoin, which doesn't seem a pipeline breaker node, we can probably just return False here or ksip the check here